### DISCUSSION TITLE
[Feature] Allow Admins via OAuth

### DISCUSSION THREAD

**Discussion by pixxon** [immich-app/immich#6683] [Feature Request]
### The feature

Hello!

I've been trying to configure Immich to be integrated with Authentik for user management. I was able to setup OAuth and register/login new users. However there are some limitations that restrict me from fully moving away from the built in user management.

I've read through the code and it seems there is no way to notify via OAuth if a user is admin or not. It would be nice if I could specify it be providing a "role" attribute with the values "Admin" or "User". ( This is similar to how Grafana handles the permissions via [role_attribute_path](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/) )

Regarding this I also saw that the user service limits the number of admins to one, but to me it seems like that limitation can be lifted rather easily. ( There are only a few references to `getAdmin()` function and most of them are test cases. )

As a side note, if the feature is something that you would like, I can contribute the implementation.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This is now implemented.
